baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/StrictUnusedVariable.java

```diff
@@ -421,61 +420,96 @@ public Boolean visitEnhancedForLoop(EnhancedForLoopTree tree, Void unused) {
     private static ImmutableList<SuggestedFix> buildUnusedParameterFixes(
             Symbol varSymbol, List<TreePath> usagePaths, VisitorState state) {
         Symbol.MethodSymbol methodSymbol = (Symbol.MethodSymbol) varSymbol.owner;
+        boolean isPrivateMethod = methodSymbol.getModifiers().contains(Modifier.PRIVATE);
         int index = methodSymbol.params.indexOf(varSymbol);
         SuggestedFix.Builder fix = SuggestedFix.builder();
         for (TreePath path : usagePaths) {
             fix.delete(path.getLeaf());
         }
-        new TreePathScanner<Void, Void>() {
-            @Override
-            public Void visitMethodInvocation(MethodInvocationTree tree, Void unused) {
-                if (getSymbol(tree).equals(methodSymbol)) {
-                    removeByIndex(tree.getArguments());
-                }
-                return super.visitMethodInvocation(tree, null);
-            }
 
-            @Override
-            public Void visitMethod(MethodTree tree, Void unused) {
-                if (getSymbol(tree).equals(methodSymbol)) {
-                    removeByIndex(tree.getParameters());
+        // Remove parameter if the method is private since we can automatically fix all invocation sites
+        // Otherwise add `_` prefix to the variable name
```

Comment:
I know this was already here, but I'm a bit worried about this behaviour.
If it's not going to auto-fix the call sites too (which is also dangerous), then it leaves the code in a state where it's harder for an IDE user to fix this (they'd have to hunt down the invocations).
On the other hand, if we don't autofix, they could just navigate to the method, and cmd+del the parameter, after which IntelliJ would autofix the callsites.
(I assume there's similar functionality in Eclipse though I haven't checked)